design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,199 @@
+in [crossplane-runtime/fieldpath] to unset the fields of a managed resource that are under `ignoreChanges`. That way
+provider maintainers could use this function in the Update method without the need to invent their own solution.
+
+Helper function would look something this (WIP):
+```go
+func UnsetIgnored(managed resource.Managed) error {
+	p, err := PaveObject(managed)
+	if err != nil {
+		return err
+	}
+	
+	for _, s := range managed.GetIgnored() {
+		err = p.DeleteField(s)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+```
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this
+new policy as an alpha feature that will be off by default and will be
+controlled by the `--enable-alpha-ignore-changes` flag in Providers.
+
+This will not prevent the field from appearing in the schema of the managed
+resources. However, we will ignore the `spec.ignoreChanges` when the feature
+is not enabled.
+
+### Implementation alternative considered
+
+An alternative was considered where the above helper function which unsets fields in the managed resource would be used
+directly in the [crossplane-runtime] reconciler, and then a version of the managed resource without the ignored fields
+would be passed to the Update method.
+
+While this solution would enable the functionality without the need to change all the providers, it has a few 
+drawbacks:
+* It doesn't allow for flexibility or handling special cases in the provider code, as the providers will just receive a 
+managed resource which does not have the ignored fields set. There could be cases where the wish is to not update the field
+on the external resource, but is useful in some other way.
+* Won't work well with Upjet as it uses Terraform underneath which would work better with its native [ignore_changes].
+
+## Solution alternative considered
+
+The idea here was to do something similar to K8s `initContainers` and add a field `initProvider` to the managed resources.
+The parameters under `initProvider` would be used only during creation.
+
+```yaml
+apiVersion: eks.aws.crossplane.io/v1alpha1
+kind: NodeGroup
+metadata:
+  name: my-group
+spec:
+  initProvider:
+    scalingConfig:
+      scalingConfig.desiredSize: 1
+  forProvider:
+    region: us-east-1
+    clusterNameRef:
+      name: sample-cluster
+    subnetRefs:
+      - name: sample-subnet1
+    nodeRoleRef:
+      name: somenoderole
+    scalingConfig:
+      maxSize: 5
+      minSize: 1
+    updateConfig:
+      maxUnavailablePercentage: 50
+      force: true
+  providerConfigRef:
+    name: example
+```
+
+The great thing about this solution is that it feels natural to add `initProvider` as we already have `forProvider`,
+`atProvider`. But on the other hand a thing that would be bothersome with this solution is that a value is needed in the 
+`initProvider`. A field cannot just be marked to be ignored, without setting the initial value.
```

Comment:
> A field cannot just be marked to be ignored, without setting the initial value.

In @turkenh's OO resources design we've extended `status.atProvider` to always contain the current values of all of an MR's fields. So presuming we had:

* `spec.initProvider` - state that is desired only at init/create time.
* `spec.forProvider` - state that is desired all the time.
* `status.atProvider` - observed actual state.

If what we're trying to say is "for this field we want to set an initial value then let the external system take over" then it seems like `spec.initProvider` covers that. If what we're trying to say is "for this field we never have desired state, just let the external system control it", wouldn't we just omit it from `spec.initProvider` _and_ `spec.forProvider` and let it only show up under `stats.atProvider`?

Is the issue (only) late initialization? i.e. That we'd need some way to say "don't late init this field", even though its not in `initProvider` or `forProvider`?

Reply:
You are right. The issue would mostly be late initialization, I need to check up a bit on late initialization to figure out how we could stop it from late init some fields. I wonder if in the end it turns into `ignoreLateInit` which brings us back to something like `ignoreChanges`

One other thing, I feel marking the field explicity ignored somewhere instead of just omitting it from `spec.initProvider` and `spec.forProvider` gives a type of "documentation" and "safety", so over time nobody decides to add this field to `spec.forProvider`.